main/design_documents/ARM_NEON_SVE_BRIDGE.rst

```diff
@@ -0,0 +1,78 @@
+=======================================
+Design Document for ARM_NEON_SVE_BRIDGE
+=======================================
+
+This document describes the design of the intrinsics defined in
+`NEON_SVE Bridge <../acle.rst#neon-sve-bridge>`_. These C intrinsics enable the
+conversion between NEON and SVE vectors with no performance overhead, and
+duplicating NEON vectors across SVE vectors.
+
+Location
+########
+
+The intrinsics are defined in a header file called 
+``arm_neon_sve_bridge.h``.
+
+This header file includes ``arm_neon.h`` and ``arm_sve.h``.
+It also defines ``__ARM_NEON_SVE_BRIDGE`` as 1.
+
+The intrinsics are defined with:
+
+.. code:: c
+
+    #ifdef __ARM_NEON_SVE_BRIDGE 1
+    //define intrinsics
+
+Suggestions & Considerations
+############################
+
+Why not enable the extraction of any 128-bit subvector in an SVE vector?
+------------------------------------------------------------------------
+
+Extracting other 128-bit subvectors goes against the no-op design.
+This functionality is already available with the ACLE intrinsic 
+``svdupq_lane``. For example:
+
+``svuint64_t sve_vector_a = ...;``
+
+``uint64x2_t neon_vec = svget_neonq_u64(svdupq_lane_u64(sve_vector_a, 1));``
+
+Why not enable the insertion of a 128-bit NEON vector into a specific SVE subvector?
+------------------------------------------------------------------------------------
+
+This behaviour is not currently available with SVE intrinsics. Such an
```

Comment:
For me, the justification is more along the lines:

> `arm_sve.h` does not currently provide any intrinsics for inserting individual elements into a specific lane of an SVE vector.  The approach should be consistent between `arm_sve.h` and `arm_neon_sve_bridge.h`.

That is, if we _did_ provide methods for inserting both individual elements and whole NEON vectors, I think the latter _would_ belong in `arm_neon_sve_bridge.h` rather than `arm_sve.h`.  But I don't think it makes sense to provide functions for whole NEON vectors when no corresponding functions exist for individual elements.

I think the nop design justification applies here too, but it's probably not worth repeating.

Reply:
done